src/commands/moderation/move.js

```diff
@@ -0,0 +1,54 @@
+const { Command, CommandError } = require('../../')
+
+module.exports = class Move extends Command {
+  constructor (client) {
+    super({
+      name: 'move',
+      aliases: ['mover'],
+      category: 'moderation',
+      requirements: { guildOnly: true, botPermissions: ['MANAGE_MESSAGES'], permissions: ['MANAGE_MESSAGES'] },
+      parameters: [{
+        type: 'messageLink', sameGuildOnly: true, forceExists: true, missingError: 'commands:move.missingMessageLink'
+      }, {
+        type: 'channel', onlySameGuild: true, acceptText: true, missingError: 'commands:move.missingChannelMention'
+      }]
+    }, client)
+  }
+
+  async run ({ channel, guild, author, t, message }, link, destinationChannel) {
+    if (!destinationChannel.permissionsFor(this.client.user.id).has('SEND_MESSAGES')) throw new CommandError(t('commands:move.cantSendInChannel'))
+
+    try {
+      if (link.channel.nsfw && !destinationChannel.nsfw) throw new CommandError(t('commands:move.channelsHaveDifferentType'))
+
+      if (!link.channel.permissionsFor(this.client.user.id).has('MANAGE_MESSAGES')) throw new CommandError(t('commands:move.cantDeleteMessage'))
+
+      if (!link.channel.permissionsFor(author.id).has('SEND_MESSAGES')) throw new CommandError(t('commands:move.userCantSend'))
+
+      const MessageObj = {}
+      let content = link.content
+
+      if (link.attachments.size >= 1) {
+        MessageObj.files = [link.attachments.first().url]
+      }
+
+      if (link.embeds.length >= 1) {
+        if (!link.channel.permissionsFor(this.client.user.id).has('EMBED_LINKS')) throw new CommandError(t('commands:move.cantSendEmbed'))
+        MessageObj.embed = {};
+        ['fields', 'title', 'description', 'url', 'timestamp', 'color', 'image', 'thumbnail', 'author'].forEach(p => {
+          MessageObj.embed[p] = link.embeds[0][p]
+        })
+      }
+
+      try {
+        await destinationChannel.send(content, MessageObj)
+        await destinationChannel.send(t('commands:move.messageMoved', { authorName: link.author.username, movedFrom: link.channel.name, movedBy: author.username }))
+        link.delete()
+      } catch (e) {
+        throw new CommandError(t('commands:move.couldntSendMessage'))
+      }
+    } catch (e) {
+      throw new CommandError(t('commands:move.couldntSendMessage'))
+    }
```

Comment:
Is that inner try/catch block necessary? Code already seems to be inside a block that returns the same error message upon error

Reply:
Let me see, I actually forgot about that.

Reply:
You're right, that additional try catch isn't needed, comitting a fix rn.

Reply:
In case you wonder why I did it, users with manage_message permission can delete a embed from a message, if that message only had the embed, then it's literally an empty message. There's a way to check that but it would use more processing than just trying to send the message. (Checking if ``content`` is '' and Object.keys(MessabeObj).length is 0)

I was kinda lazy and dind't notice that we already had that try catch.

Reply:
Cool